### DISCUSSION TITLE
How to make SSR request without cache?

### DISCUSSION THREAD

**Discussion by gvalex13** [nuxt/nuxt#17633] [Questions] [answered]
I need to get a fresh data every time a visitor returns to the page. That is I need a regular SSR request, without caching. But it seems there is no such option for useAsyncData/useFetch. Am I right?

I've also tried this scenario on page loading: 
```
refreshNuxtData(key)
const result = await useAsyncData(key, nuxtApp => nuxtApp.$api(url, apiOptions), asyncDataOptions)
```

but it did not work either  - I got 500 error on the first load (b/c there is no cache yet?) and duplicates requests when I return to page on client side (b/c refreshNuxtData init data refreshing?)

Is it possible to make a regular request without caching with SSR? Or how to get fresh data on enter the page?

At the moment the only way I found is using $fetch directly. But it works client side only.

**Comment by Vi-jay** [comment]
i need this too，The fetch option api can be used in nuxt@v2 but not in v3

**Comment by Vi-jay** [comment]
I also solved by this
```ts
export function genAsyncDataKey(key) {
  if (process.server) return key;
  return key + Math.random().toFixed(32).slice(2, 10);
}
const {pending,refresh} = await useLazyAsyncData(genAsyncDataKey("Test"),async () => {
  console.log("exec");
  await sleep(2)
  return 1;
});

```

**Reply by gvalex13** [reply]
I tried to generate a unique key also. But it seems should be the same on server and client side. When they are different I receive empty value from useAsyncData on client side. Try to receive some data from API and display it with your example. 
So using a unique key for every request did not work for me too

**Reply by Vi-jay** [reply]
> I tried to generate a unique key also. But it seems should be the same on server and client side. When they are different I receive empty value from useAsyncData on client side. Try to receive some data from API and display it with your example. So using a unique key for every request did not work for me too

You can do this, store the state in the store, it will make the server and client render the same state the first time, but this is not a perfect solution, it can only be regarded as a trick

```ts
const indexStore = useIndexStore();
const { pending } = await useLazyAsyncData(key, async () => {
  await sleep(2);
  indexStore.count++;
  return Promise.resolve();
});
console.log(indexStore.count);
```

**Reply by gvalex13** [reply]
Thank you for solution, buy you are right, it can be regarded as a trick only. We still need a way to control cache

**Comment by gvalex13** [comment, marked as answer]
I seemed to miss the option that solves the issue:
`useAsyncData(key, nuxtApp => nuxtApp.$api(url), { initialCache: false })`
And it WAS in the documentation. My fault

**Comment by Vi-jay** [comment]
Yes, you are right. Thanks for solving this problem
